### ISSUE TITLE
Git repository history might contain bloat

### ISSUE THREAD

**Issue by BottledByte** [vegastrike/Vega-Strike-Engine-Source#83]
Currently, size of this repo is around 150 MB and size of Assets-Production repo is around 1.4 GB. However, when git `clone`-ing those repos, full history included (~1.5 GB without it), *approx. 3.4 GB is downloaded*. ~800 MB of this is VS's engine history and another ~1.1 GB is from the asset repo's history.

Vega Strike has many years of development behind it, so it can be said that the size of engine's history is okay. However, assets repo might be a problem, as it contains all the models, textures, etc. Git treats those files badly (and those files tend to be quite big), so if it is not a problem now, it might become one in the future (if those assets would get updated several times).

Also, this issue somewhat relates to #38. From those 150 MB of actual engine repo size... ~90% is Boost. If I understood it correctly, Boost was not originally kept in the SVN repo, only as a reference. But Git makes snapshots of everything, so it is probable that it keeps snapshots of entire Boost in history and if the Boost would be upgraded (and vendored the same way as is now), another snapshot would be created (creating a lot of bloat).

@nabaco suggested that a notice is added to readme that the size of history is somewhat big and maybe suggest there to use shallow git `clone`.

Feel free to close this if you do not consider it an issue. (and rather a Git feature :) )

**Comment by BenjamenMeyer** [comment]
Yes, asset repo is large and will grow that can't be helped too much.
Third party libs like boost need a better vendoring solution than being simply included in the repository.

**Comment by evertvorster** [comment]
I just built two packages for Arch Linux AUR for vegastrike. Both of them pull the full git repo before building the package. One for the engine, one for the data. 

Download size for the engine was around 700mbm which is a bit vulgar for something that boils down to 7mb package once built. 

The data one can understand that it is big, and that is around a 1gb download.

The way the AUR works is that anyone that wants to install this package would have to download both git repos as well, so a little de-bloating here would be much appreciated.

**Comment by BenjamenMeyer** [comment]
@evertvorster our goal is that they won't have to download the git, but can have two packages: (1) engine (small) and (2) data, with the data package being used to support multiple games - we already know of 2 separate games (Under the Coldest Sea, provided by our Asset Production repo) and the Privateer WCU Parallel Universe (provided by the PWCU GH Org). Let's continue that discussion in the #131 issue.

**Comment by stephengtuggy** [comment]
I would say that the Git repository history _almost certainly_ contains bloat -- a lot of it. When I clone only my fork of the repo, it is about a GB smaller -- even with the data / assets. Then as soon as I add the upstream remote and fetch that, my working copy grows by at least a GB again.

**Comment by BenjamenMeyer** [comment]
If anyone has a suggestion for commands that can be put in the README to help people with this then please open a PR.
This should only effect developers and folks building from source. We should produce artifacts for platforms of interest for each release that can be downloaded and used by our users so they shouldn't have to be concerned about it.

I'll leave this open a bit longer, then we'll close it out.

**Comment by BenjamenMeyer** [comment]
```
git clone git@github.com:vegastrike/Vega-Strike-Engine-Source.git --shallow-since=2023-09-27
```
Produces a 22 MB download. I'll file a PR shortly for documenting this.
